gcore/gdaldrivermanager.cpp

```diff
@@ -1146,6 +1219,496 @@ void GDALDriverManager::ReorderDrivers()
+}
+
+static const char *const apszProxyMetadataItems[] = {
+    GDAL_DMD_LONGNAME,
+    GDAL_DMD_EXTENSIONS,
+    GDAL_DMD_EXTENSION,
+    GDAL_DCAP_RASTER,
+    GDAL_DCAP_MULTIDIM_RASTER,
+    GDAL_DCAP_VECTOR,
+    GDAL_DCAP_GNM,
+    GDAL_DMD_OPENOPTIONLIST,
+    GDAL_DCAP_OPEN,
+    GDAL_DCAP_CREATE,
+    GDAL_DCAP_CREATE_MULTIDIMENSIONAL,
+    GDAL_DCAP_CREATECOPY,
+    GDAL_DMD_SUBDATASETS,
+    GDAL_DCAP_MULTIPLE_VECTOR_LAYERS,
+    GDAL_DCAP_NONSPATIAL,
+    GDAL_DMD_CONNECTION_PREFIX,
+    GDAL_DCAP_VECTOR_TRANSLATE_FROM,
+};
+
+const char *GDALPluginDriverProxy::GetMetadataItem(const char *pszName,
+                                                   const char *pszDomain)
+{
+    const auto IsListedProxyMetadataItem = [](const char *pszItem)
+    {
+        for (const char *pszListedItem : apszProxyMetadataItems)
+        {
+            if (EQUAL(pszItem, pszListedItem))
+                return true;
+        }
+        return false;
+    };
+
+    if (!pszDomain || pszDomain[0] == 0)
+    {
+        if (EQUAL(pszName, "IS_NON_LOADED_PLUGIN"))
+        {
+            return !m_poRealDriver ? "YES" : nullptr;
+        }
+        else if (EQUAL(pszName, "MISSING_PLUGIN_FILENAME"))
+        {
+            return m_osPluginFullPath.empty() ? m_osPluginFileName.c_str()
+                                              : nullptr;
+        }
+        else if (IsListedProxyMetadataItem(pszName))
+        {
+            const char *pszValue =
+                GDALDriver::GetMetadataItem(pszName, pszDomain);
+            if (!pszValue && EQUAL(pszName, GDAL_DMD_EXTENSION))
+            {
+                const char *pszOtherValue =
+                    GDALDriver::GetMetadataItem(GDAL_DMD_EXTENSIONS, pszDomain);
+                if (pszOtherValue && strchr(pszOtherValue, ' '))
+                    return pszOtherValue;
+            }
+            else if (!pszValue && EQUAL(pszName, GDAL_DMD_EXTENSIONS))
+            {
+                return GDALDriver::GetMetadataItem(GDAL_DMD_EXTENSION,
+                                                   pszDomain);
+            }
+            return pszValue;
+        }
+        else if (!EQUAL(pszName, GDAL_DMD_PLUGIN_INSTALLATION_MESSAGE) &&
+                 m_oSetMetadataItems.find(pszName) != m_oSetMetadataItems.end())
+        {
+            return GDALDriver::GetMetadataItem(pszName, pszDomain);
+        }
+    }
+
+    auto poRealDriver = GetRealDriver();
+    if (!poRealDriver)
+        return nullptr;
+    return poRealDriver->GetMetadataItem(pszName, pszDomain);
+}
+
+/************************************************************************/
+/*                           GetRealDriver()                            */
+/************************************************************************/
+
+GDALDriver *GDALPluginDriverProxy::GetRealDriver()
+{
+    if (m_osPluginFullPath.empty())
+        return nullptr;
+
+    CPLMutexHolderD(&hDMMutex);
```

Comment:
should the mutex be moved before the  m_osPluginFullPath.empty() check?

Reply:
no, this is not needed.

I've just added the following comment:
```
    // No need to take the mutex has this member variable is not modified
    // under the mutex.
```